### PR TITLE
Serialization

### PR DESCRIPTION
**Pull request by MattTaylorDLS**
Change to using annotypes for all Generators, Mutators, Excluders and ROIS.
Makes serialization a lot simpler, as it's handled automatically by annotypes.

### CODE REVIEW

File: scanpointgenerator/rois/circular_roi.py [hunk 1]
```diff
@@ -12,21 +12,30 @@
...
+
 
 @ROI.register_subclass("scanpointgenerator:roi/CircularROI:1.0")
 class CircularROI(ROI):
 
     def __init__(self, centre, radius):
+        # type: (UCentre, ARadius) -> None
         super(CircularROI, self).__init__()
 
         if radius == 0.0:
```
Comment (thomascobb) [inline scanpointgenerator/rois/circular_roi.py]:
```suggestion
        if radius <= 0.0:
```

File: scanpointgenerator/rois/polygonal_roi.py [hunk 2]
```diff
@@ -11,29 +11,32 @@
 #
 ###
 
+from annotypes import Anno, Union, Array, Sequence
+
 from scanpointgenerator.core import ROI
 from scanpointgenerator.compat import range_, np
 
+with Anno("x positions for polygon vertices"):
+    APoints_X = Array[np.float64]
```
Comment (thomascobb) [inline scanpointgenerator/rois/polygonal_roi.py]:
```suggestion
    APointsX = Array[np.float64]
```

Comment (thomascobb) [reply scanpointgenerator/rois/polygonal_roi.py]: Mixing CamelCase_With_Underscores looks a bit odd...

**Comment by xen-0** [comment]
GDA is going to have to be taught about annotypes.

**Pull request merged**